force-app/main/default/classes/ExtensionControllerTest.cls

```diff
@@ -0,0 +1,142 @@
+        System.assert(assertCon.Is_Primary_Contact__c == true);
+        System.assertEquals('/apex/PrimaryContact?id='+refAcct.Id, ref.getUrl());
+
+    }
+    
+    //Testing search() method
+    @isTest 
+    private static void testsearchSuccess()
+    {   
+        initData('testMethodNameSuccess');
+        Account searchAcc = [SELECT Id, Name FROM Account Where Name = 'TestAccount'];
+        
+        List<Contact> Cons = [Select Id, LastName, Is_Primary_Contact__c FROM Contact WHERE LastName Like '%TestCon%'];
+        Map<Integer, Contact> ContMap = new Map<Integer, Contact>();
+                for(Integer index = 0 ; index < Cons.size(); index++) {
+                    ContMap.put(index, Cons[index]);}
+         
+        String sText = 'TestC';
+        
+        PageReference searchPage = Page.PrimaryContact; 
+               Test.setCurrentPage(searchPage);
+               searchPage.getParameters().put('Id', String.valueOf(searchAcc.Id));
+               searchPage.getParameters().put('searchText',sText);
+               
+        //create a standard controller and pass the dummy account controller
+        ApexPages.StandardController accountController = new ApexPages.StandardController(searchAcc);
+
+        //Get the Standard Controller and pass it to controller extension
+        extentionController ext = new extentionController(accountController);
+        
+        //setting ContactMap to dummy values
+        ext.ContactMap = ContMap;
+        
+        //Test
+        Test.startTest();
+        
+        PageReference ref = ext.search();
+        
+        Test.stopTest();
+        
+        //Assert
+        System.assertEquals(null, ref);
+    }
+    
+    //Test that will fail
+     @isTest
+    private static void testMethodNameFailure() {
+        initData('testMethodNameFailure');
+        Account failAct = [SELECT Id, Name FROM Account WHERE Name = 'FailAcc'];
+        
+        String sText = 'TestC';        
+        PageReference failPage = Page.PrimaryContact; 
+               Test.setCurrentPage(failPage);
+               failPage.getParameters().put('Id', String.valueOf(failAct.Id));
+               failPage.getParameters().put('searchText',sText);
+        
+        //create a standard controller and pass the dummy account controller
+        ApexPages.StandardController accountController = new ApexPages.StandardController(failAct);
+
+        //Get the Standard Controller and pass it to controller extension
+        extentionController ext = new extentionController(accountController);
+        
+        //Setting ContactMap to null
+        ext.ContactMap = null;
+        
+        //Feeding with null Map<Integer, Contact>
+        Test.startTest();
+        
+        ext.search();
+        
+        Test.stopTest();
+        
+        //Assertion
+        System.assertEquals('/apex/PrimaryContact?id='+failAct.Id, ext.search().getUrl());
```

Comment:
please assert result, not just url of page.
for example your number of contacts could be 0 (or list null)